### PR TITLE
Added preliminary local run before call and deploy

### PR DESCRIPTION
**Pull request by SilkovAlexander**
tonos-cli now can execute the transaction locally for deploy and call commands before executing it onchain. If local execution fails, onchain execution is not performed. Local run can be enabled by setting the flag `local_run` in the tonos-cli config. Default value for this config is `false`. 
Added print of code hash to account command.

### CODE REVIEW

File: src/call.rs [hunk 1]
```diff
@@ -245,14 +244,40 @@ async fn query_account_boc(ton: TonClient, addr: &str) -> Result<String, String>
...
+                boc: state_boc,
+                unlimited_balance: None,
+            },
+            ..Default::default()
+        },
+    )
+    .await;
+
+    if res.is_err() {
+        return Err(format!("Local execution has failed:\n{:#}", res.err().unwrap()));
```
Comment (Keshoid) [inline src/call.rs:268]: I tried it out and in my opinion we should remove this text: `Local execution has failed:\n{:#}`, just return original error.

**Review by Keshoid — APPROVED** [review]
LGTM

File: src/call.rs [hunk 2]
```diff
@@ -377,17 +380,47 @@ pub async fn call_contract_with_result(
...
         }
 
+        let mut retry: bool = true;
+        let error_handler = |err: ClientError| {
+            // obtaining error code
+            let code = err.code.clone();
+            // but if it was simulated locally and local exit code is not zero,
+            // we ignore previous exit code because it means we shouldn't make a retry.
+            if !err.data["exit_code"].is_null() {
+                if err.data["exit_code"].as_i64().unwrap() != 0 {
```
Comment (Keshoid) [inline src/call.rs]: unwrap_or_default()

File: src/call.rs [hunk 3]
```diff
@@ -377,17 +380,47 @@ pub async fn call_contract_with_result(
...
+                    retry = false;
+                }
+            }
+            // There is also another way how SDK can print local run results.
+            let local_error = err.data["local_error"]["data"]["exit_code"].clone();
+            if !local_error.is_null() {
+                if local_error.as_i64().unwrap() != 0 {
```
Comment (Keshoid) [inline src/call.rs]: unwrap_or_default()

**Review by Keshoid — APPROVED** [review]

File: src/call.rs (hunk 1 above)
Comment (SilkovAlexander) [reply src/call.rs]: Fixed

File: src/call.rs (hunk 2 above)
Comment (SilkovAlexander) [reply src/call.rs]: Fixed

File: src/call.rs (hunk 3 above)
Comment (SilkovAlexander) [reply src/call.rs]: Fixed

**Review by Keshoid — APPROVED** [review]
LGTM

**Pull request merged**